visitor.php

```diff
@@ -364,32 +478,140 @@ private function addArrayHashNotation(Doc $docComment): ?Doc
         );
 
-        return new Doc($newDocComment, $docComment->getLine(), $docComment->getFilePos());
+        return new Doc($newDocComment, $docComment->getStartLine(), $docComment->getStartFilePos());
     }
 
-    private function addAdditionalParams(Doc $docComment): ?Doc
+    /**
+     * @param array<int, WordPressTag> $additions
+     * @return array<int, WordPressTag>
+     */
+    private function discoverInheritedArgs(DocBlock $docblock, array $additions): array
+    {
+        foreach ($additions as $add) {
+            if ($add->tag === '@phpstan-param') {
+                // We can't yet handle merging args.
+                return $additions;
+            }
+        }
+
+        /** @var Param[] $params */
+        $params = $docblock->getTagsByName('param');
+
+        foreach ($params as $param) {
+            $additions = array_merge($additions, $this->getInheritedTagsForParam($param));
+        }
+
+        return $additions;
+    }
+
+    /**
+     * @return array<int, WordPressTag>
+     */
+    private function getInheritedTagsForParam(Param $param): array
+    {
+        $type = $param->getType();
+
+        if ($type === null) {
+            return [];
+        }
+
+        $typeName = self::getTypeNameFromType($type);
+
+        if ($typeName === null) {
+            return [];
+        }
+
+        $paramDescription = $param->getDescription();
+
+        if ($paramDescription === null) {
+            return [];
+        }
+
+        list($description) = explode("\n\n", $paramDescription->__toString());
+
+        if (strpos($description, '()') === false) {
+            return [];
+        }
+
+        $description = str_replace("\n", ' ', $description);
+        $additions = [];
+
+        foreach ($this->additionalTags as $symbolName => $tags) {
+            $search = sprintf(
+                'see %s()',
+                $symbolName
+            );
+
+            if (stripos($description, $search) === false) {
+                continue;
+            }
+
+            $match = self::getMatchingInheritedTag($param, $tags, $symbolName);
+
+            if ($match !== null) {
+                $additions[] = $match;
+            }
+        }
+
+        return $additions;
+    }
+
+    /**
+     * @param array<int, WordPressTag> $tags
+     */
+    private static function getMatchingInheritedTag(Param $param, array $tags, string $symbolName): ?WordPressTag
+    {
+        $paramName = $param->getVariableName();
+        $matchNames = [
+            $paramName,
+            'args',
+            'options',
+            'query',
+        ];
+        $matchingTags = array_filter($tags, static function(WordPressTag $tag) use ($matchNames): bool {
+            return in_array($tag->name, $matchNames, true);
+        });
+
+        foreach ($matchingTags as $tag) {
+            $addTag = clone $tag;
+            $addTag->name = $paramName;
+            $addTag->description = sprintf(
+                '@see %s()',
```

Comment:
This is the greatest feature of this PR! (no sarcasm)

Please remove the `@` sign as it is a tag-in-a-tag.

Reply:
```suggestion
                'See %s()',
```